### DISCUSSION TITLE
Error on sending MQTT

### DISCUSSION THREAD

**Discussion by javrolijk** [timmbogner/Farm-Data-Relay-System#183] [Q&A] [answered]
Hi All,

I recently changed the library files to the latest, and now I'm receiving "Error on sending MQTT" after one/two messages have been sent to my MQTT broker located on a Raspberry Pi. I'm testing out, a Lilygo LoRa ESP32 OLED sending sensor packets to another LoRa ESP32 OLED connected through serial to an ESP32-WROOM module which is connected to the MQTT broker.

Not sure how to troubleshoot this as I'm seeing data getting to the MQTT broker, however only once. Any suggestions are highly appreciated.

![image](https://github.com/timmbogner/Farm-Data-Relay-System/assets/70583691/56117d10-b88f-4296-9093-00583a4e10d5)

**Comment by timmbogner** [comment]
I'm away from my computer and workshop for a few more days, but I'll investigate when I'm home. My first order of business will be to fix and re-do some of the MQTT functionality.

**Reply by javrolijk** [reply]
@timmbogner 
As @aviateur17 indicated, the issue was fixed by increasing the MQTT default (256) buffer size to 1024.
I've revised the 'fdrs_gateway_mqtt.h' as follows:
`#define MQTT_MAX_BUFF_SIZE 1024

void begin_mqtt()
{
    client.setServer(mqtt_server, mqtt_port);
	client.setBufferSize(MQTT_MAX_BUFF_SIZE);
	client.setBufferSize(1024);
    if (!client.connected())
    {
        reconnect_mqtt(5);
    }
    client.setCallback(mqtt_callback);
}

void mqtt_publish(const char *payload)
{
    if (!client.publish(TOPIC_DATA, payload))
    {
        DBG(" Error on sending MQTT");
#if defined(USE_SD_LOG) || defined(USE_FS_LOG)
        sendLog();
#endif
    }
    else
    {
#if defined(USE_SD_LOG) || defined(USE_FS_LOG)
        if (last_log_write >= last_mqtt_success)
        {
            releaseLogBuffer();
            resendLog();
        }
        time(&last_mqtt_success);
#endif
    }
}`

**Reply by timmbogner** [reply]
Awesome, glad you got it going. I'll use this as the basis for the fix and give you and @aviateur17 credit once I get back to things!

**Reply by timmbogner** [reply]
@javrolijk I added a commit to fix this a while back but forgot to mention it!

**Comment by aviateur17** [comment, marked as answer]
The issue is based on the default MQTT buffer size of 256 bytes.   The default needs to be adjusted upwards to accommodate larger amounts of data.   I'll look up and link to a previous comment on the same in a bit.  Should probably submit a PR just to increase the MQTT buffer size.
